torch/_inductor/pcache.py

```diff
@@ -0,0 +1,317 @@
+from concurrent.futures import Future, ThreadPoolExecutor
+from functools import cached_property
+from os import getenv
+from pathlib import Path
+from typing_extensions import Self
+
+from torch.utils._filelock import FileLock
+
+
+# Key represents data used to query from cache instances
+Key = str
+# Value represents data stored in/retrieved from cache instances
+Value = bytes
+
+
+class Cache:
```

Comment:
Shouldn't this subclass Generic with defaults of the Key and Value TypeAliases? Would make it more flexible if we need to change it later

Reply:
Done; I haven't used Generic previously so please double check my understanding. I ended up using TypeVar as Generic does not take TypeAlias